src/lib/structures/game/Game.ts

```diff
@@ -226,68 +220,40 @@ export class Game {
 		}
 	}
 
-	// TODO: More inclusive regexes
-	public validateAnswer(guess: string) {
-		switch (this.acceptedAnswer) {
-			case AcceptedAnswer.Song: {
-				return this.validateSong(guess);
-			}
-
-			case AcceptedAnswer.Artist: {
-				return this.validateArtist(guess);
-			}
-
-			case AcceptedAnswer.Either: {
-				return this.validateArtist(guess) || this.validateSong(guess);
-			}
-
-			case AcceptedAnswer.Both: {
-				if (this.guessedThisRound) {
-					return this.guessedThisRound === AcceptedAnswer.Song ? this.validateArtist(guess) : this.validateSong(guess);
-				}
-
-				const guessedArtist = this.validateArtist(guess);
-				if (guessedArtist) {
-					this.guessedThisRound = AcceptedAnswer.Artist;
-					return true;
-				}
-
-				const guessedSong = this.validateSong(guess);
-				if (guessedSong) {
-					this.guessedThisRound = AcceptedAnswer.Song;
-					return true;
-				}
+	// will potentially make these not abstract and put common behaviour here
+	public abstract guess(guessMessage: Message): void;
+	public abstract onTrackEnd(): void;
+	public abstract guessAnswer(guess: string, user: User): void;
 
+	// returns true if new player guesses the artist
+	protected guessArtist(guess: string, user: User) {
+		for (const artist of Object.keys(this.roundData.trackArtistsGuessed)) {
+			if (this.roundData.trackArtistsGuessed[artist].has(user)) {
 				return false;
 			}
+			const match = guess === artist || jaroWinkler(guess, artist) >= kGuessThreshold;
+			if (match) {
+				this.roundData.trackArtistsGuessed[artist].add(user);
+				return true;
```

Comment:
Unlike the comment says, this returns true if the player guesses *a* artist. However, there's no indication of which artist is guessed, and in the trackEnd function, it only looks at the "primary artist". Could you explain your train of thought a bit more?

Reply:
Alright, could you add that in a comment either above the line I commented or the method itself?